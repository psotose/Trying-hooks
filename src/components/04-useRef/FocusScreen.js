import React, { useRef } from 'react';
import '../01-useState/counter.css';

export const FocusScreen = () => {

  const inputRef = useRef();

  const handleClick = () => {
    // document.querySelector('input').select();
    inputRef.current.select();
  };

  return (
    <div>
      <h1>Focus Screen</h1>
      <hr />

      <input
        ref={ inputRef }
        className="form-control"
        placeholder="Su nombre"
      />

      <button 
        className="btn btn-success mt-3"
        onClick={ handleClick }
      > 
        Focus 
      </button>  
    </div>
  )
}
