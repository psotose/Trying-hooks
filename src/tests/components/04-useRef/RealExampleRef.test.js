import React from 'react';
import { shallow } from 'enzyme';
import { RealExampleRef } from '../../../components/04-useRef/RealExampleRef';

describe('Testing <RealExampleRef />', () => {
  const wrapper = shallow( <RealExampleRef /> );
  
  test('should render correctly', () => {
    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('MultipleCustomHooks').exists() ).toBe(false);
  });

  test('should show <MultipleCustomHooks /> component', () => {

    wrapper.find('button').simulate('click');
    expect( wrapper.find('MultipleCustomHooks').exists() ).toBe(true);
  });
})