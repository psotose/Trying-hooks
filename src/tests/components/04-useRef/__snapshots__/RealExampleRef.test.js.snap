// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <RealExampleRef /> should render correctly 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
